### PR TITLE
Add disk metrics script and template

### PR DESCRIPTION
**Pull request by hallipr**
emits metrics like:

```
logmetric: {"name":"agent_driveinfo_size_bytes","value":296559366144,"timestamp":"2021-12-07T00:10:39.783592+00:00","labels":{"name":"C:\\","volumeLabel":"System","driveType":"Fixed","driveFormat":"NTFS"}}
logmetric: {"name":"agent_driveinfo_available_bytes","value":176327012352,"timestamp":"2021-12-07T00:10:39.7865546+00:00","labels":{"name":"C:\\","volumeLabel":"System","driveType":"Fixed","driveFormat":"NTFS"}}
logmetric: {"name":"agent_driveinfo_size_bytes","value":214748360704,"timestamp":"2021-12-07T00:10:39.7888924+00:00","labels":{"name":"D:\\","volumeLabel":"Data","driveType":"Fixed","driveFormat":"NTFS"}}
logmetric: {"name":"agent_driveinfo_available_bytes","value":194948788224,"timestamp":"2021-12-07T00:10:39.7912437+00:00","labels":{"name":"D:\\","volumeLabel":"Data","driveType":"Fixed","driveFormat":"NTFS"}}
```

and on linux:
```
logmetric: {"name":"agent_driveinfo_size_bytes","value":34078720,"timestamp":"2021-12-07T00:14:38.5586226+00:00","labels":{"name":"/snap/snapd/13640","volumeLabel":"/snap/snapd/13640","driveType":"Fixed","driveFormat":"squashfs"}}
logmetric: {"name":"agent_driveinfo_available_bytes","value":0,"timestamp":"2021-12-07T00:14:38.5590331+00:00","labels":{"name":"/snap/snapd/13640","volumeLabel":"/snap/snapd/13640","driveType":"Fixed","driveFormat":"squashfs"}}
logmetric: {"name":"agent_driveinfo_size_bytes","value":157457793024,"timestamp":"2021-12-07T00:14:38.5594543+00:00","labels":{"name":"/mnt","volumeLabel":"/mnt","driveType":"Fixed","driveFormat":"lofs"}}
logmetric: {"name":"agent_driveinfo_available_bytes","value":144993136640,"timestamp":"2021-12-07T00:14:38.5598729+00:00","labels":{"name":"/mnt","volumeLabel":"/mnt","driveType":"Fixed","driveFormat":"lofs"}}
logmetric: {"name":"agent_driveinfo_size_bytes","value":44302336,"timestamp":"2021-12-07T00:14:38.5602935+00:00","labels":{"name":"/snap/snapd/14066","volumeLabel":"/snap/snapd/14066","driveType":"Fixed","driveFormat":"squashfs"}}
logmetric: {"name":"agent_driveinfo_available_bytes","value":0,"timestamp":"2021-12-07T00:14:38.5607022+00:00","labels":{"name":"/snap/snapd/14066","volumeLabel":"/snap/snapd/14066","driveType":"Fixed","driveFormat":"squashfs"}}
```

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]
Looks reasonable.

File: eng/common/scripts/Write-FileSystemMetrics.ps1 [hunk 1]
```diff
@@ -0,0 +1,29 @@
+foreach($volume in Get-Volume) {
```
Comment (benbp) [inline eng/common/scripts/Write-FileSystemMetrics.ps1:1]: Do you plan to add cross-platform disk metric support?

Comment (weshaggard) [reply eng/common/scripts/Write-FileSystemMetrics.ps1:1]: This should be cross-platform. Is there something you think should not work?

Comment (hallipr) [reply eng/common/scripts/Write-FileSystemMetrics.ps1:1]: I didn't realize Get-Volume was windows only.   Looking at `df` for linux

Comment (hallipr) [reply eng/common/scripts/Write-FileSystemMetrics.ps1]: Added using .net core's `[IO.DriveInfo]::GetDrives()`

**Review by weshaggard — APPROVED** [review]
Looks reasonable. Lets be sure to test this on windows, linux, and mac.

File: eng/common/pipelines/templates/steps/write-filesystemmetrics.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+steps:
+    - task: Powershell@2
+      inputs:
+        filePath: $(Build.SourcesDirectory)/eng/common/scripts/Write-FileSystemMetrics.ps1
+        pwsh: true
+      displayName: Write filesystem metrics
```
Comment (sima-zhu) [inline eng/common/pipelines/templates/steps/write-filesystemmetrics.yml:6]: The step is to log disk space, so it is better to fail silently. 
Better to add `continueOnError` here.

**Comment by benbp** [comment]
Agree with @sima-zhu's comment, otherwise LGTM.

**Review by benbp — APPROVED** [review]

File: eng/common/pipelines/templates/steps/write-filesystemmetrics.yml (hunk 2 above)
Comment (hallipr) [reply eng/common/pipelines/templates/steps/write-filesystemmetrics.yml:6]: Added, thanks

**Pull request merged**